Source/WebKit/UIProcess/win/PageClientImpl.cpp

```diff
@@ -232,6 +240,63 @@ void PageClientImpl::preferencesDidChange()
     notImplemented();
 }
 
+bool PageClientImpl::handleRunOpenPanel(WebPageProxy*, WebFrameProxy*, const FrameInfoData&, API::OpenPanelParameters* parameters, WebOpenPanelResultListenerProxy* listener)
+{
+    ASSERT(parameters);
+    ASSERT(listener);
+
+    HWND viewWindow = viewWidget();
+    if (!IsWindow(viewWindow))
+        return false;
+
+    bool isAllowMultipleFiles = parameters->allowMultipleFiles();
+    Vector<wchar_t> fileBuf(isAllowMultipleFiles ? maxFilePathsListSize : MAX_PATH);
+
+    OPENFILENAME ofn { };
+
+    // Need to zero out the first char of fileBuf so GetOpenFileName doesn't think it's an initialization string
+    fileBuf[0] = '\0';
+
+    ofn.lStructSize = sizeof(ofn);
+    ofn.hwndOwner = viewWindow;
+    ofn.lpstrFile = fileBuf.data();
+    ofn.nMaxFile = fileBuf.size();
+    auto dialogTitle = uploadFileText();
+    auto dialogTitleCharacters = dialogTitle.wideCharacters(); // Retain buffer long enough to make the GetOpenFileName call
+    ofn.lpstrTitle = dialogTitleCharacters.data();
+    ofn.Flags = OFN_ENABLESIZING | OFN_FILEMUSTEXIST | OFN_PATHMUSTEXIST | OFN_EXPLORER;
+    if (isAllowMultipleFiles)
+        ofn.Flags |= OFN_ALLOWMULTISELECT;
+
+    if (GetOpenFileName(&ofn)) {
+        auto files = fileBuf.data();
+        Vector<String> fileList;
+        String file(files);
+        if (isAllowMultipleFiles) {
+            while (!file.isEmpty()) {
+                // When using the OFN_EXPLORER flag, the file list is null delimited.
+                // When you create a String from a ptr to this list, it will use strlen to look for the null character.
+                // Then we find the next file path string by using the length of the string we just created.
+                wchar_t* nextFilePtr = files + file.length() + 1;
+                String nextFile(nextFilePtr);
+                // If multiple files are selected, there will be a directory name first, which we don't want to add to the vector.
+                // We know a single file was selected if there is only one filename in the list.  
+                // In that case, we don't want to skip adding the first (and only) name.
+                if (files != fileBuf.data() || nextFile.isEmpty())
+                    fileList.append(file);
+                files = nextFilePtr;
+                file = nextFile;
+            }
```

Comment:
This loop should look like:
~~~c++
for (auto p = fileBuffer.data(); *p;) {
  auto length = wcslen(p);
  String filename(p, length);
  (...)
  p += length + 1;
}
~~~

Reply:
When multiple files are selected, the directory and file names are given separately.
I found that the previous code did not work correctly.
I'll change the code based on the following information.
https://stackoverflow.com/questions/26317556/how-to-get-list-of-selected-files-when-using-getopenfilename-with-multiselect

Reply:
Changed the around codes at https://github.com/WebKit/WebKit/commit/70ce9358458f9401040946ef3392f2ae55d6fa3e